src/main/java/org/folio/services/migration/BatchedReadStream.java

```diff
@@ -0,0 +1,112 @@
+package org.folio.services.migration;
+
+import io.vertx.core.Handler;
+import io.vertx.core.streams.ReadStream;
+import java.util.ArrayList;
+import java.util.List;
+
+public class BatchedReadStream<T> implements ReadStream<List<T>> {
+  private static final int DEFAULT_BATCH_SIZE = 100;
+
+  private final ReadStream<T> delegate;
+  private final int batchSize;
+  private final List<T> buffer;
+
+  private Handler<List<T>> handler;
+  private Handler<Throwable> exceptionHandler;
+
+  public BatchedReadStream(ReadStream<T> delegate) {
+    this(delegate, DEFAULT_BATCH_SIZE);
+  }
+
+  public BatchedReadStream(ReadStream<T> delegate, int batchSize) {
+    this.delegate = delegate;
+    this.batchSize = batchSize;
+    this.buffer = new ArrayList<>(batchSize);
+  }
+
+  @Override
+  public BatchedReadStream<T> exceptionHandler(Handler<Throwable> handler) {
+    this.exceptionHandler = handler;
+    delegate.exceptionHandler(handler);
+    return this;
+  }
+
+  @Override
+  public BatchedReadStream<T> handler(Handler<List<T>> handler) {
+    if (handler == null) {
```

Comment:
so, is there a way to clear the handler property of the class?

Reply:
Thanks, changed. Now handler is also cleared on delegate.